src/quick-lint-js/fe/parse-statement.cpp

```diff
@@ -2015,6 +2017,20 @@ void parser::parse_and_visit_function_parameters(parse_visitor_base &v,
               .init_kind = variable_init_kind::normal,
               .first_parameter_begin = first_parameter_begin,
           });
+
+      if (parameter->kind() == expression_kind::optional ||
+          (parameter->kind() == expression_kind::type_annotated &&
+           parameter->child_0()->kind() == expression_kind::optional)) {
+        previous_optional_span = parameter->span();
+      } else {
+        if (previous_optional_span.has_value()) {
+          this->diag_reporter_->report(
+              diag_optional_parameter_cannot_be_followed_by_required_parameter{
+                  .optional_parameter = *previous_optional_span,
+                  .required_parameter = parameter->span()});
+        }
+        previous_optional_span = std::nullopt;
```

Comment:
What's the purpose of this code? If I delete it, your tests still pass.

Reply:
I added this to prevent reflagging subsequent required parameters for the same optional one. Especially now that the optional parameter is highlighted it seemed more appropriate to trigger the diagnostic only once. I added a test to illustrate this. Let me know if you'd rather I remove the test and this line.